nest: Fix error for operators without CRDs

SamiSousa opened the pull request:
- Add checks for object pathes in nest.py before accessing values
- Add test case for test_nest.py for operator without CRDs

Fixes an issue seen in CI on this PR: https://github.com/operator-framework/community-operators/pull/278
The CI runs `operator-courier nest` on operators, but for this operator it errors out with the obscure message`'customresourcedefinitions'`. This fix adds more field checks into the nest function, and handles operators without owned CRDs.
CI log: https://travis-ci.com/operator-framework/community-operators/builds/108167507#L454

SamiSousa commented:
/cc @kevinrizza @jeremylinlin Please take a look, thank you!

jeremylinlin commented on operatorcourier/nest.py:
```diff
@@ -58,18 +71,23 @@ def nest_bundles(yaml_files, registry_dir, temp_registry_dir):
-            else:
-                errors.append("CRD %s mentioned in CSV %s was not found in directory."
-                              % (crd_name, csv_name))
+        if "customresourcedefinitions" in csv["spec"]:
```
`KeyError` here if `spec` is not defined?

SamiSousa replied:
I append an error earlier on for this, but it's true that this would throw a `KeyError`
Should I break early if a key is missing, or should I continue to aggregate errors?

SamiSousa replied:
I'm leaning towards using `continue` whenever a required key isn't found

jeremylinlin commented on operatorcourier/nest.py:
```diff
@@ -46,7 +52,14 @@ def nest_bundles(yaml_files, registry_dir, temp_registry_dir):
     for csv in csvs:
+        if "metadata" not in csv:
+            errors.append("CSV has no `metadata` field defined")
         csv_name = csv["metadata"]["name"]
```
What about the `name` field?

SamiSousa replied:
Good catch, will add a check here

jeremylinlin commented on operatorcourier/nest.py:
```diff
@@ -46,7 +52,14 @@ def nest_bundles(yaml_files, registry_dir, temp_registry_dir):
+            errors.append("CSV %s has no `spec` field defined" % csv_name)
+        if "version" not in csv["spec"]:
+            errors.append("CSV %s has no `spec.version` field defined" % csv_name)
         version = csv["spec"]["version"]
```
`KeyError` here if `spec` or `version` is not defined?

jeremylinlin commented on operatorcourier/nest.py:
```diff
@@ -58,18 +71,23 @@ def nest_bundles(yaml_files, registry_dir, temp_registry_dir):
+                    if "name" not in csv_crd:
+                        errors.append("CSV %s has an owned CRD without a `name` field"
+                                      "defined" % csv_name)
+                    crd_name = csv_crd["name"]
```
Same issue as above. I guess without using an `else` condition, the code will still try to access the `name` field when it is not present and will throw an error here?

SamiSousa commented:
@kevinrizza Addressed your comment, separated this PR into two commits relating to the distinct changes.
@jeremylinlin Addressed your KeyError concerns by adding `continue` statements in the CSV loop to avoid accessing keys that don't exist.
PTAL

jeremylinlin commented:
@SamiSousa Thanks for addressing the comments. It looks good!

jeremylinlin commented:
/lgtm

kevinrizza commented on tests/test_files/bundles/nest/bundle2/svcat.package.yaml:
```diff
@@ -0,0 +1,5 @@
+channels:
+- name: alpha
+  currentCSV: svcat.v0.1.34
+      
```
This is missing a newline at the end of the file.

SamiSousa replied:
Ah, I left some whitespace on line 5, will fix this. Thank you!

SamiSousa commented:
@kevinrizza Addressed your comment, thank you!

kevinrizza commented:
/lgtm
